### PR TITLE
[Critical] Revamped the structure of data depending upon branches, years and semesters

### LINKED ISSUE #6: Segregation of study materials depending on semesters
Have a separate page for spring and autumn semesters of every year and branch.

### PR DESCRIPTION
**Pull request by EmperorYP7**
Fixes #6

### CODE REVIEW

**Review by sandesh-08 — APPROVED** [review]
LGTM!!!
Great work @EmperorYP7

File: studentportal/src/data/branches.json [hunk 1]
```diff
@@ -2,49 +2,185 @@
     {
         "id": 1,
         "name": "First Year CSE",
-        "intro":"Learn important subjects of Second Year Computer Science and Engineering course like Operating Systems, Data Structures, Graph Theory, Microprocessor and Interfacing,etc which will help you to learn and grow more.",
-        "link":"/first-year-cse"
+        "branch":"CSE",
+        "semester":[
+            { 
+                "id":"CSE0101",
```
Comment (EmperorYP7) [inline studentportal/src/data/branches.json:8]: This is so nice to have! 🤩 Great work @tusharjain0022

**Review by EmperorYP7 — COMMENTED** [review]
LGTM! Well done @tusharjain0022! 🙌🏻

**Review by ShashankS1ngh — APPROVED** [review]
LGTM 👍

**Review by Sahil-Gupta582 — APPROVED** [review]

**Review by sourav1kumar — APPROVED** [review]
LGTM 🥇

**Review by Abhayraj1707 — APPROVED** [review]
LGTM !! 🤩🤩
Nice work @EmperorYP7

**Review by tusharjain0022 — APPROVED** [review]

**Comment by tusharjain0022** [comment]
@all-contributors please add @EmperorYP7  for code.

**Pull request merged**